index.ts

```diff
@@ -1,139 +1,199 @@
+        compilerOptions.paths || (compilerOptions.baseUrl && nonRelative)
+      );
+
+      if (
+        typeof importer === 'undefined' ||
+        importee.startsWith('\0') ||
+        !enabled
+      ) {
+        return null;
+      }
+
+      const hasMatchingPath =
+        !!compilerOptions.paths &&
+        Object.keys(compilerOptions.paths).some((path) =>
+          new RegExp('^' + path.replace('*', '.+') + '$').test(importee)
+        );
+
+      if (!hasMatchingPath && !nonRelative) {
+        return null;
+      }
+
+      if (importee.startsWith('.')) {
+        return null; // never resolve relative modules, only non-relative
+      }
+
+      const { resolvedModule } = nodeModuleNameResolver(
+        importee,
+        importer,
+        compilerOptions,
+        sys
+      );
+
+      if (!resolvedModule) {
+        return null;
+      }
+
+      const { resolvedFileName } = resolvedModule;
+
+      if (!resolvedFileName || resolvedFileName.endsWith('.d.ts')) {
+        return null;
+      }
+
+      const targetFileName = path.join(
+        outDir,
+        preserveExtensions
+          ? resolvedFileName
+          : resolvedFileName.replace(/\.tsx?$/i, '.js')
+      );
+
+      const resolved = absolute
+        ? sys.resolvePath(targetFileName)
+        : targetFileName;
+
+      return transform ? transform(resolved) : resolved;
+    },
+  };
 };
 
 const getTsConfig = (configPath?: string): TsConfig => {
-	const defaults: TsConfig = { compilerOptions: {}, outDir: '.' };
-
-	if (!configPath) {
-		return defaults;
-	}
-
-	const configJson = sys.readFile(configPath);
-
-	if (!configJson) {
-		return defaults;
-	}
-
-	const { config } = parseConfigFileTextToJson(configPath, configJson);
+  const defaults: TsConfig = { compilerOptions: { outDir: '.' } };
+  if (typeof configPath !== 'string') {
+    return defaults;
+  }
+
+  // Read in tsconfig.json
+  const configJson = sys.readFile(configPath);
+  if (configJson == null) {
+    return defaults;
+  }
+
+  const { config: rootConfig } = parseConfigFileTextToJson(
+    configPath,
+    configJson
+  );
+  const rootConfigWithDefaults = {
+    ...rootConfig,
+    ...defaults,
+    compilerOptions: {
+      ...defaults.compilerOptions,
+      ...(rootConfig.compilerOptions ?? {}),
+    },
+  };
+  const resolvedConfig = handleTsConfigExtends(
+    rootConfigWithDefaults,
+    configPath
+  );
+
+  return resolvedConfig;
+};
 
-	return { ...defaults, ...config };
+const handleTsConfigExtends = (
+  config: TsConfig,
+  rootConfigPath: string
+): TsConfig => {
+  if (!('extends' in config) || typeof config.extends !== 'string') {
+    return config;
+  }
+
+  let extendedConfigPath;
+  try {
+    // Try to resolve as a module (npm)
+    extendedConfigPath = require.resolve(config.extends);
+  } catch (e) {
+    // Try to resolve as a file relative to the current config
+    extendedConfigPath = path.join(
+      path.dirname(rootConfigPath),
+      config.extends
```

Comment:
`extends` can actually be an array since TS 5.0. I'm pretty sure there's probably a better method from the TypeScript API to retrieve the tsconfig.

I wrote this a long time ago when I was still completely unfamiliar with the API 😅

Reply:
https://stackoverflow.com/a/70013087/2897426 maybe this is the correct solution

Reply:
that actually worked quite good :)